parcels/particle.py

```diff
@@ -206,10 +206,14 @@ def execute(self, pyfunc=AdvectionRK4, time=None, dt=1., timesteps=1,
             output_steps = timesteps
         timeleaps = int(timesteps / output_steps)
         # Execute kernel in sub-stepping intervals (leaps)
-        current = time or self.grid.time[0]
+        if self.grid.U.dtsign < 0 and self.grid.V.dtsign < 0:
```

Comment:
Again, adding the `reverse=False` default argument to execute would make this line a little cleaner.